storage-bigtable/src/access_token.rs

```diff
@@ -91,24 +92,35 @@ impl AccessToken {
 
     /// Call this function regularly to ensure the access token does not expire
     pub async fn refresh(&self) {
-        // Check if it's time to try a token refresh
+        //Check if it's time to try a token refresh
         {
             let token_r = self.token.read().unwrap();
+
             if token_r.1.elapsed().as_secs() < token_r.0.expires_in() as u64 / 2 {
+                info!("Token not ready to be refreshed");
                 return;
             }
+            warn!("Token ready to be refreshed");
+            warn!("Current Token: {:#?}", self.token);
 
             #[allow(deprecated)]
             if self
                 .refresh_active
                 .compare_and_swap(false, true, Ordering::Relaxed)
             {
                 // Refresh already pending
+                let wait_time: u64 = 2;
+                let wait_time_millis = std::time::Duration::from_millis(wait_time * 1000);
+                warn!("Refresh already pending... waiting {} seconds before trying again...", wait_time);
+
+                thread::sleep(wait_time_millis);
```

Comment:
```suggestion
                tokio::time::sleep(wait_time_millis).await;
```

to avoid blocking the executor